### ISSUE TITLE
Exception happened during the evaluation of VCOCO

### ISSUE THREAD

**Issue by GWwangshuo** [fredzzhang/spatio-attentive-graphs#31]
Hi, I trained your method on `VCOCO` dataset and generated the `vcoco_results.pkl`. However, when I ran the utilities provided by Gupta, I got the exception below:
![image](https://user-images.githubusercontent.com/6948631/105137222-f232bf00-5b2d-11eb-9e8c-8774ebf2ac07.png)

Could you please kindly help me handling this problem? Thanks a lot. Besides, I also attached the code for evaluation of mine:
```python
from vsrl_eval import VCOCOeval
from collections import defaultdict

class CacheTemplate(defaultdict):
    """A template for VCOCO cached results """
    def __init__(self, **kwargs):
        super().__init__()
        for k, v in kwargs.items():
            self[k] = v
    def __missing__(self, k):
        seg = k.split('_')
        # Assign zero score to missing actions
        if seg[-1] == 'agent':
            return 0.
        # Assign zero score and a tiny box to missing <action,role> pairs
        else:
            return [0., 0., .1, .1, 0.]



if __name__ == "__main__":
    vsrl_annot_file = "data/vcoco/vcoco_val.json"
    coco_file = "data/instances_vcoco_all_2014.json"
    split_file = "data/splits/vcoco_val.ids"
    det_file = "spatio-attentive-graphs/vcoco_cache/vcoco_results.pkl"
    vcocoeval = VCOCOeval(vsrl_annot_file, coco_file, split_file)
    vcocoeval._do_eval(det_file, ovr_thresh=0.5)
```

**Comment by fredzzhang** [comment]
Hi, @GWwangshuo 

The error you had was caused by performing `np.amax` on an empty array. I'm not very familiar with the details of the evaluation code unfortunately. But judging from the code you posted, it seems like you are using the wrong partition.
```python
vsrl_annot_file = "data/vcoco/vcoco_val.json"
...
split_file = "data/splits/vcoco_val.ids"
```
The detection results were generated on the test set, but here you are evaluating against the validation set.

Don't know if that could solve your problem.

Cheers,
Fred

**Comment by GWwangshuo** [comment]
@fredzzhang Thanks for your reply. Problem solved. Close this issue.
